### PR TITLE
return from write

### LINKED ISSUE #67: Support result value from write transaction callbacks
A single value should be possible to be returned as a result of the write transaction callback.
var value = realm.write(() {})

### PR DESCRIPTION
**Pull request by desistefanova**
Return result from write method callback. Fixes #67

### CODE REVIEW

**Review by nielsenko — APPROVED** [review]

File: lib/src/realm_class.dart [hunk 1]
```diff
@@ -166,11 +166,12 @@ class Realm {
   ///
   /// If no exception is thrown from within the callback, the transaction will be committed.
   /// It is more efficient to update several properties or even create multiple objects in a single write transaction.
-  void write(void Function() writeCallback) {
+  TResult write<TResult>(TResult Function() writeCallback) {
```
Comment (blagoev) [inline lib/src/realm_class.dart:169]: I suggest we use `T` here.

Comment (blagoev) [inline lib/src/realm_class.dart:169]:
```suggestion
  T write<T>(T Function() writeCallback) {
```

File: test/realm_test.dart [hunk 2]
```diff
@@ -595,4 +595,18 @@ Future<void> main([List<String>? args]) async {
     expect(mainSchools[0].branches[0].students.length + mainSchools[0].branches[1].students.length, 3);
     realm.close();
   });
+
+  test('Realm write returns added object', () {
```
Comment (blagoev) [inline test/realm_test.dart:599]:
```suggestion
  test('Realm write returns result', () {
```

**Review by blagoev — CHANGES_REQUESTED** [review]

File: lib/src/realm_class.dart (hunk 1 above)
Comment (desistefanova) [reply lib/src/realm_class.dart]: done

**Review by blagoev — APPROVED** [review]

**Pull request merged**